### PR TITLE
Remove licensor component

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Remove deprecated component related to self-hosted

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/16981

## Release Notes
```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [x] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/admin/License.tsx [hunk 1]
```diff
@@ -33,11 +33,7 @@ export default function License() {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
-    const featureList = license?.enabledFeatures;
-    const features = license?.features;
-
-    // if user seats is 0, it means that there is no user limit in the license
-    const userLimit = license?.seats === 0 ? "Unlimited" : license?.seats;
+    const userLimit = "Unlimited";
```
Comment (AlexTugarev) [inline components/dashboard/src/admin/License.tsx:36]: ❓ do we need the License page at all? 
cc. @geropl and @atduarte

We can do that as a follow-up.

File: components/server/ee/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -334,38 +331,7 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
...
-            userCount = cachedUserCount;
-        }
-
-        if (!this.licenseEvaluator.isEnabled(feature, userCount)) {
-            throw new ResponseError(ErrorCodes.EE_LICENSE_REQUIRED, "enterprise license required");
-        }
-    }
-
     async validateLicense(ctx: TraceContext): Promise<LicenseValidationResult> {
-        const v = this.licenseEvaluator.validate();
```
Comment (AlexTugarev) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:354]: 👍🏻  not removing the method should work fine during roll out.

File: yarn.lock [hunk 3]
```diff
@@ -12522,22 +12444,6 @@ node-forge@^0.10.0:
   resolved "https://registry.npmjs.org/node-forge/-/node-forge-0.10.0.tgz"
   integrity sha512-PPmu8eEeG9saEUvI97fm4OYxXVB6bFvyNTyiUOBichBpFG8A1Ljw3bY62+5oOjDEMHRnd0Y7HQ+x7uzxOzC6JA==
 
-node-gyp@^8.3.0:
```
Comment (AlexTugarev) [inline yarn.lock:12525]: 🕺🏻

**Review by AlexTugarev — APPROVED** [review]
LGTM

/hold just in case if this needs a second reviewer

**Comment by AlexTugarev** [comment]
❓ What should happen with the license handling in `./gitpod/.werft`?

**Comment by geropl** [comment]
@AlexTugarev I have no time to look into this PR, but: We still rely on there being a license/empty license for the "Dedicated" setup atm; this is what helps determine the BillingMode for Dedicated (just like self-hosted) to be `none`.

Could you check that with this PR this is still the case? :pray:

**Comment by geropl** [comment]
We just double-checked: There is no usage of `config.license` anymore. :heavy_check_mark:

**Comment by geropl** [comment]
/unhold

Deployment is :green_circle:

**Pull request merged**